tasks/modify_settings.yml

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Configure setting with value
+  when: item.previous | d() != 'replaced'
+  vars:
+    __bootloader_with_value: >-
+      {{ item.name is defined and item.value is defined }}
+    __bootloader_setting: >-
+      {{ __bootloader_with_value | ternary(item.name + '=' + item.value | d(),
+      item.name) }}"
+    __bootloader_absent: "{{ item.state | d('present') == 'absent' }}"
+  block:
+    - name: Check boot setting {{ __bootloader_setting }}
+      shell: >-
+        set -euo pipefail;
+        grubby --info=DEFAULT | grep '^args="' | sed 's/^args=//'
+      register: __bootloader_check_setting
+      changed_when: false
+
+    - name: Configure boot setting {{ __bootloader_setting }}
+      command: >-
+        grubby
+        --update-kernel=DEFAULT
+        {{ __bootloader_absent | ternary('--remove-args=', '--args=')
+        }}{{ __bootloader_setting | quote }}
+      changed_when: true
+      when: >-
+        ((__bootloader_setting not in __bootloader_check_setting.stdout) | bool)
+        == ((not __bootloader_absent) | bool)
```

Comment:
I guess this works because the set of valid kernel parameters is pretty small, and there aren't any parameters which have a string which is a subset of another parameter e.g. if the kernel command line is `foobar bizbuz` and `__bootloader_setting` is `buz`, then this will report that `buz` is already set, which is erroneous.

Can `grubby --args` report that it set something?  Ideally, `grubby --args=buz` would report 0, `grubby --args=bizbuz` would report 1 (already set), and `grubby --args=bad_parameter` would report 2 (error).  Then we could use `register: __grubby_result` and `failed_when: __grubby_result.rc not in [0, 1]` and `changed_when: __grubby_result.rc == 0`, or something like that.

If not, then I think we need to implement some sort of simple parser which splits the existing args into a `list` or `dict` that we can use to test exact membership.  I'm not sure of the syntax of the kernel command line - it looks like space delimited, but I have no idea if parameters like `kernel_param="value with spaces"` is allowed.  Maybe our bootloader dev contact knows, or maybe just play around with `grubby` and see what happens with `--args=key=value\ with\ spaces`

Reply:
Grubby arguments are simple, they are other `value` or `key=value`. Grubby does not return anything after adding or removing vars, rc is always 0.

Note that grubby also does not validate arguments, so you can put anything there, grub will then ignore arguents that do not make sense for it.

`value\ with\ spaces` is not supported, if I do `--args=key=value\ with\ spaces` grubby would add `value\ with\ spaces` as a separate attributes.